save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/FossGraphView.kt

```diff
@@ -52,10 +55,37 @@ val fossGraphView: VFC = VFC {
 val fossGraph: FC<FossGraphViewProps> = FC { props ->
     useBackground(Style.WHITE)
 
+    val projectWindowOpenness = useWindowOpenness()
+
     val params = useParams()
     val vulnerabilityName = params["vulnerabilityName"]!!.toString()
 
+    val (vulnerabilityProjects, setVulnerabilityProjects) = useState<Set<VulnerabilityProjectDto>>(setOf())
     val (vulnerability, setVulnerability) = useState(VulnerabilityDto.empty)
+    val (isUpdateVulnerability, setIsUpdateVulnerability) = useState(false)
+
+    val fetchProject: (VulnerabilityProjectDto) -> Unit = { project ->
+        setVulnerability {
+            it.copy(projects = it.projects.plus(project))
+        }
+        setVulnerabilityProjects {
+            it.plus(project)
+        }
```

Comment:
Do you want to duplicate `VulnerabilityProject`s info?

Reply:
Fixed

Reply:
_Vulnerability_ is needed to display on the frontend, and _VulnerabilityProjects_ is passed to the backend.